types/react-redux/index.d.ts

```diff
@@ -15,12 +15,12 @@ type Store<S> = Redux.Store<S>;
 type Dispatch<S> = Redux.Dispatch<S>;
 type ActionCreator<A> = Redux.ActionCreator<A>;
 
-export interface DispatchProp<T> {
-  dispatch: Dispatch<T>
+export interface DispatchProp<S> {
+  dispatch: Dispatch<S>;
 }
 
-interface ComponentDecorator<TMergedProps> {
-    <TOwnProps>(component: Component<(TOwnProps & TMergedProps) | TOwnProps>): ComponentClass<TOwnProps>;
+interface ComponentDecorator<TMergedProps, TOwnProps> {
+    <T extends TOwnProps>(component: Component<T & TMergedProps>): ComponentClass<T>;
```

Comment:
Also, FWIW, it is possible to make your code work today with `connect()<OwnProps>(Component)`, but I would like to improve the default behaviour for everyone involved.